tests/bud.bats

```diff
@@ -33,6 +33,12 @@ load helpers
   expect_output --substring "hello"
 }
 
+#Verify https://github.com/containers/buildah/issues/4312
+@test "multi-stage: verify heading arg in base image name" {
+  # stage `FROM stage_${my_env}` must be resolved with default arg value and build should be successful.
+  run_buildah build $WITH_POLICY_JSON -t source -f $BUDFILES/multi-stage-builds/Dockerfile.arg_in_stage
```

Comment:
This could be merged with the "bud with ARG before FROM" test, because they seem to be two halves of the same test.

Reply:
SGTM amended.